### PR TITLE
ci: Go 1.17 related changes

### PR DESCRIPTION
**Pull request by ernestojeda**
Changes related to Go 1.17 and cache optimizations

### CODE REVIEW

File: cmd/support-scheduler/Dockerfile [hunk 1]
```diff
@@ -16,7 +16,7 @@
 # 
 #  ----------------------------------------------------------------------------------
 
-ARG BUILDER_BASE=golang:1.16-alpine3.14
+ARG BUILDER_BASE=golang:1.17-alpine3.15
```
Comment (lenny-intel) [inline cmd/support-scheduler/Dockerfile:19]: Other changes missing?

**Review by lenny-intel — CHANGES_REQUESTED** [review]
Also need to update go version used by snap here: https://github.com/edgexfoundry/edgex-go/blob/b0e5e643282aa8d6ca4ca25954737fd68a04a1f7/snap/snapcraft.yaml#L584

**Review by lenny-intel — APPROVED** [review]
LGTM

**Review by lenny-intel — APPROVED** [review]

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -581,7 +581,7 @@ parts:
     # see comment for static-packages part about specifying a source part here
     source: snap/local/build-helpers
     build-snaps:
-      - go/1.16/stable
+      - go/1.17/stable
```
Comment (farshidtz) [inline snap/snapcraft.yaml:584]: This doesn't look right.

**Review by farshidtz — CHANGES_REQUESTED** [review]

File: snap/snapcraft.yaml (hunk 2 above)
Comment (farshidtz) [reply snap/snapcraft.yaml:584]: I was too slow. It is already fixed..

Comment (ernestojeda) [reply snap/snapcraft.yaml:584]: yeah, had a bad find/replace. 🥺

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**